src/model/test/CoilCoolingDXVariableRefrigerantFlowFluidTemperatureControl_GTest.cpp

```diff
@@ -0,0 +1,150 @@
+  EXPECT_TRUE(coil.setIndoorUnitEvaporatingTemperatureFunctionofSuperheatingCurve(curve));
+
+  Schedule schedule = coil.availabilitySchedule();
+  boost::optional<ScheduleConstant> _scheduleConstant = schedule.optionalCast<ScheduleConstant>();
+  ASSERT_TRUE(_scheduleConstant);
+  EXPECT_EQ((*_scheduleConstant).value(), 0.5);
+  ASSERT_TRUE(coil.ratedTotalCoolingCapacity());
+  EXPECT_EQ(1, coil.ratedTotalCoolingCapacity().get());
+  EXPECT_FALSE(coil.isRatedTotalCoolingCapacityAutosized());
+  EXPECT_FALSE(coil.autosizedRatedTotalCoolingCapacity());
+  ASSERT_TRUE(coil.ratedSensibleHeatRatio());
+  EXPECT_EQ(2, coil.ratedSensibleHeatRatio().get());
+  EXPECT_FALSE(coil.autosizedRatedSensibleHeatRatio());
+  EXPECT_EQ(3, coil.indoorUnitReferenceSuperheating());
+  Curve _curve = coil.indoorUnitEvaporatingTemperatureFunctionofSuperheatingCurve();
+  boost::optional<CurveCubic> indoorUnitEvaporatingTemperatureFunctionofSuperheatingCurve = _curve.optionalCast<CurveCubic>();
+  EXPECT_TRUE(indoorUnitEvaporatingTemperatureFunctionofSuperheatingCurve);
+
+  coil.autosizeRatedTotalCoolingCapacity();
+  coil.autosizeRatedSensibleHeatRatio();
+
+  EXPECT_FALSE(coil.ratedTotalCoolingCapacity());
+  EXPECT_TRUE(coil.isRatedTotalCoolingCapacityAutosized());
+  EXPECT_FALSE(coil.autosizedRatedTotalCoolingCapacity());
+  EXPECT_FALSE(coil.ratedSensibleHeatRatio());
+  EXPECT_TRUE(coil.isRatedSensibleHeatRatioAutosized());
+  EXPECT_FALSE(coil.autosizedRatedSensibleHeatRatio());
+}
+
+TEST_F(ModelFixture, CoilCoolingDXVariableRefrigerantFlowFluidTemperatureControl_Clone) {
+  Model model;
+
+  CoilCoolingDXVariableRefrigerantFlowFluidTemperatureControl coil(model);
+
+  EXPECT_TRUE(coil.setRatedTotalCoolingCapacity(10));
+
+  CoilCoolingDXVariableRefrigerantFlowFluidTemperatureControl coilClone =
+    coil.clone(model).cast<CoilCoolingDXVariableRefrigerantFlowFluidTemperatureControl>();
+  ASSERT_TRUE(coilClone.ratedTotalCoolingCapacity());
+  EXPECT_EQ(10, coilClone.ratedTotalCoolingCapacity().get());
+
+  Model model2;
+  CoilCoolingDXVariableRefrigerantFlowFluidTemperatureControl coilClone2 =
+    coil.clone(model2).cast<CoilCoolingDXVariableRefrigerantFlowFluidTemperatureControl>();
+  ASSERT_TRUE(coilClone2.ratedTotalCoolingCapacity());
+  EXPECT_EQ(10, coilClone2.ratedTotalCoolingCapacity().get());
+}
+
+TEST_F(ModelFixture, CoilCoolingDXVariableRefrigerantFlowFluidTemperatureControl_Remove) {
+  Model model;
+  auto size = model.modelObjects().size();
+  CoilCoolingDXVariableRefrigerantFlowFluidTemperatureControl coil(model);
+  EXPECT_EQ(1u, model.getObjectsByType(CurveQuadratic::iddObjectType()).size());
+  EXPECT_FALSE(coil.remove().empty());
+  EXPECT_EQ(0u, model.getObjectsByType(CurveQuadratic::iddObjectType()).size());
+  EXPECT_EQ(0u, model.getObjectsByType(CoilCoolingDXVariableRefrigerantFlowFluidTemperatureControl::iddObjectType()).size());
+  EXPECT_EQ(size + 2, model.modelObjects().size());  // Always On Discrete, OnOff
```

Comment:
Number of curves? IN case of the clone, please check the Inlet / Outlet Nodes to make sure they are reset (comment applies to the other coil)

Reply:
How do we check nodes here?

Reply:
hum, I think I am quite confused by the fact that the `CoilCooling/HeatingDXVariableRefrigerantFlow`s have a Coil Inlet/Outlet Node in the IDD. I wonder why... since it doesn't seem to be used anywhere. Was that some kind of future-planning?